src/clover/analysis.py

```diff
@@ -340,36 +344,38 @@ def plot_outputs(  # pylint: disable=too-many-locals, too-many-statements
         plt.ylabel("Energy generation / kWh per day")
         plt.title("Daily energy generation of 1 kWp of solar capacity")
         plt.savefig(
-            os.path.join(figures_directory, "solar_output_yearly.png"), transparent=True
+            os.path.join(figures_directory, "solar_output_yearly.png"),
+            transparent=True,
         )
         plt.close()
         pbar.update(1)
-
-        # Plot the grid availablity profile.
-        if grid_profile is not None:
-            reshaped_data = np.reshape(
-                grid_profile.iloc[0:HOURS_PER_YEAR].values, (365, 24)
-            )
-            heatmap = sns.heatmap(
-                reshaped_data, vmin=0, vmax=1, cmap="Greys_r", cbar=False
-            )
-            heatmap.set(
-                xticks=range(0, 24, 2),
-                xticklabels=range(0, 24, 2),
-                yticks=range(0, 365, 30),
-                yticklabels=range(0, 365, 30),
-                xlabel="Hour of day",
-                ylabel="Day of year",
-                title="Grid availability of the selected profile.",
-            )
-            plt.xticks(rotation=0)
-            plt.tight_layout()
-            plt.savefig(
-                os.path.join(figures_directory, "grid_availability_heatmap.png"),
-                transparent=True,
-            )
-            plt.close()
-            pbar.update(1)
+        
+        for grid_profile in grid_profiles:
```

Comment:
@BenWinchester Can you please check that and tell me if it is correct?!

Reply:
It looks good, though we'll want to add the length of the grid profiles to our loop, though this is just cosmetic :smile:

Reply:
Ben this section is giving me the following error same as in simulation:
![image](https://user-images.githubusercontent.com/106335872/176020074-3fb5b2e9-4ba2-4a40-9726-213fab5e4d55.png)

Not very comfortable with the plotting to be honest... but I need it to understand if everything is working fine in the simulation